### PR TITLE
Add pipeline to add uPheno relations more directly

### PR DESCRIPTION
**Pull request by matentzn**
The weird problem with this pipeline right now is massive runtime. I am experimenting but cant tell exactly what is causing the problem..

### CODE REVIEW

File: src/ontology/phenio.Makefile [hunk 1]
```diff
@@ -91,18 +75,6 @@ $(ONT)-relation-graph.tsv: $(MINIMAL_PATH)
 			--obo-prefixes true \
 			--verbose true
 
-# base-plus. No externally imported axioms, but reasoning is performed.
-$(ONT)-base-plus.owl: $(EDIT_PREPROCESSED) $(OTHER_SRC) $(IMPORT_FILES)
```
Comment (matentzn) [inline src/ontology/phenio.Makefile:95]: Not needed (no base-plus release for PHENIO, base = base-plus)

File: src/ontology/phenio.Makefile [hunk 2]
```diff
@@ -26,43 +26,27 @@ $(TMPDIR)/$(ONT)-full-unreasoned.owl: $(SRC) $(OTHER_SRC)
...
-			 annotate --ontology-iri $(URIBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) \
-			 convert -o $@.tmp.owl && mv $@.tmp.owl $@
+### Download ROBOT plugin for upheno
+$(ROBOT_PLUGINS_DIRECTORY)/upheno.jar:
+	mkdir -p $(ROBOT_PLUGINS_DIRECTORY)
+	curl -L -o $@ https://github.com/monarch-initiative/monarch-robot-plugins/releases/download/v0.0.1/monarch-robot-extensions-0.0.1.jar
+
+$(ONT)-full.owl: $(TMPDIR)/$(ONT)-full-unreasoned.owl | all_robot_plugins
+	$(ROBOT) merge --input $< \
+		upheno:extract-upheno-relations --root-phenotype UPHENO:0001001 --relation UPHENO:0000003 --relation UPHENO:0000001 \
```
Comment (matentzn) [inline src/ontology/phenio.Makefile]: This is the key line, right now works but is super slow.

**Comment by matentzn** [comment]
@caufieldjh Can you test this before merging it? But also be very dilligent in reviewing :D

**Comment by caufieldjh** [comment]
Will take a look.
Do you expect the changes to increase overall runtime?

**Comment by matentzn** [comment]
No; maybe 10 minutes more. But my guess would be 10 minutes less.

**Comment by caufieldjh** [comment]
Most of the build time is occupied by relationgraph - on the GCP instance I'm building this on, nearly 4 hours is just relationgraph work

**Comment by caufieldjh** [comment]
OK, I think this is working as expected.
Example:
```owl
  <!-- http://purl.obolibrary.org/obo/MP_0000266 -->

    <owl:Class rdf:about="http://purl.obolibrary.org/obo/MP_0000266">
        <owl:equivalentClass>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/BFO_0000051"/>
                <owl:someValuesFrom>
                    <owl:Class>
                        <owl:intersectionOf rdf:parseType="Collection">
                            <rdf:Description rdf:about="http://purl.obolibrary.org/obo/PATO_0000051"/>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002314"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UBERON_0000948"/>
                            </owl:Restriction>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002573"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/PATO_0000460"/>
                            </owl:Restriction>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/UPHENO_9000005"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UPHENO_MP"/>
                            </owl:Restriction>
                        </owl:intersectionOf>
                    </owl:Class>
                </owl:someValuesFrom>
            </owl:Restriction>
        </owl:equivalentClass>
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/MP_0002127"/>
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/UPHENO_0076810"/>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/BFO_0000051"/>
                <owl:someValuesFrom>
                    <owl:Class>
                        <owl:intersectionOf rdf:parseType="Collection">
                            <rdf:Description rdf:about="http://purl.obolibrary.org/obo/PATO_0000051"/>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002314"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UBERON_0000948"/>
                            </owl:Restriction>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002573"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/PATO_0000460"/>
                            </owl:Restriction>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/UPHENO_9000005"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UPHENO_MP"/>
                            </owl:Restriction>
                        </owl:intersectionOf>
                    </owl:Class>
                </owl:someValuesFrom>
            </owl:Restriction>
        </rdfs:subClassOf>
    </owl:Class>
```

**Comment by caufieldjh** [comment]
And the linked UPHENO class:
```owl
    <!-- http://purl.obolibrary.org/obo/UPHENO_0076810 -->

    <owl:Class rdf:about="http://purl.obolibrary.org/obo/UPHENO_0076810">
        <owl:equivalentClass>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/BFO_0000051"/>
                <owl:someValuesFrom>
                    <owl:Class>
                        <owl:intersectionOf rdf:parseType="Collection">
                            <rdf:Description rdf:about="http://purl.obolibrary.org/obo/PATO_0000051"/>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002314"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UBERON_0000948"/>
                            </owl:Restriction>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002573"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/PATO_0000460"/>
                            </owl:Restriction>
                        </owl:intersectionOf>
                    </owl:Class>
                </owl:someValuesFrom>
            </owl:Restriction>
        </owl:equivalentClass>
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/UPHENO_0076776"/>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/BFO_0000051"/>
                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/OBA_VT0000266"/>
            </owl:Restriction>
        </rdfs:subClassOf>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/BFO_0000051"/>
                <owl:someValuesFrom>
                    <owl:Class>
                        <owl:intersectionOf rdf:parseType="Collection">
                            <rdf:Description rdf:about="http://purl.obolibrary.org/obo/PATO_0000051"/>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002314"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UBERON_0000948"/>
                            </owl:Restriction>
                            <owl:Restriction>
                                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002573"/>
                                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/PATO_0000460"/>
                            </owl:Restriction>
                        </owl:intersectionOf>
                    </owl:Class>
                </owl:someValuesFrom>
            </owl:Restriction>
        </rdfs:subClassOf>
    </owl:Class>
```

**Review by caufieldjh — APPROVED** [review]

**Comment by balhoff** [comment]
> Most of the build time is occupied by relationgraph - on the GCP instance I'm building this on, nearly 4 hours is just relationgraph work

@caufieldjh I can't say it will make a difference, but it doesn't look like the heap size is specified for relation-graph in this workflow. I recommend setting that to something close to the max available using `JAVA_OPTS`. (e.g. `JAVA_OPTS="-Xmx16G -XX:+UseParallelGC"` (that garbage collector is also probably faster too)

**Comment by caufieldjh** [comment]
Thanks @balhoff ! I think the heap size gets set in the runfile here: https://github.com/monarch-initiative/phenio/blob/30bf6ba7657e99b4150fc2063cd78fbb097b696a/src/ontology/run.sh#L58
But then that just gets passed as a ROBOT arg so I'll need to see if relation-graph is picking it up.
Probably need to specify that garbage collector arg too.

**Pull request merged**